### PR TITLE
feat: UIInputResult, UIDropdownResult and UICanvasInformation implementation

### PR DESCRIPTION
**Pull request by sandrade-dcl**
### What this PR includes?
- Implemention of `OnValueChanged` and `OnSubmit` events for `UIInput` SDK component.
- Implemention of `OnValueChanged` event for `UIDropdown` SDK component.
- Implemention of the `UICanvasInformation` SDK component.

#### UICanvasInformation
- `UICanvasInformationSystem`

### Other things included
- `UIInputComponent` class has been refactorized in order to make it as poolable class and remove the intermediate class `DCLInputText` that was unneeded.
- `UIDropdownComponent` class has been refactorized in order to make it as poolable class and remove the intermediate class `DCLDropdown` that was unneeded.

### CODE REVIEW

File: Explorer/Assets/DCL/SDKComponents/SceneUI/Systems/UIBackground/UIBackgroundInstantiationSystem.cs [hunk 1]
```diff
@@ -93,13 +94,22 @@ private void LoadUIBackgroundTexture(ref PBUiBackground sdkModel, ref UIBackgrou
...
+                if (uiBackgroundComponent.TexturePromise.Value.TryConsume(World, out StreamableLoadingResult<Texture2D> promiseResult))
+                {
+                    // Backgrounds with texture
+                    UiElementUtils.SetupDCLImage(ref uiBackgroundComponent.Image, ref sdkModel, promiseResult.Asset);
+                    uiBackgroundComponent.Status = LifeCycle.LoadingFinished;
+                }
+            }
+            catch (Exception e)
+            {
+                Debug.LogError($"Error consuming texture promise: {e.Message}");
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/SDKComponents/SceneUI/Systems/UIBackground/UIBackgroundInstantiationSystem.cs]: Please use `ReportHub` instead

File: Explorer/Assets/DCL/SDKComponents/SceneUI/Systems/UIBackground/UIBackgroundInstantiationSystem.cs [hunk 2]
```diff
@@ -93,13 +94,22 @@ private void LoadUIBackgroundTexture(ref PBUiBackground sdkModel, ref UIBackgrou
...
+            try
+            {
+                if (uiBackgroundComponent.TexturePromise.Value.TryConsume(World, out StreamableLoadingResult<Texture2D> promiseResult))
+                {
+                    // Backgrounds with texture
+                    UiElementUtils.SetupDCLImage(ref uiBackgroundComponent.Image, ref sdkModel, promiseResult.Asset);
+                    uiBackgroundComponent.Status = LifeCycle.LoadingFinished;
+                }
+            }
+            catch (Exception e)
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/SDKComponents/SceneUI/Systems/UIBackground/UIBackgroundInstantiationSystem.cs]: Why do you catch an exception here?
I think (unless you expect `SetupDCLImage` to throw) you should be checking `promiseResult.Succeeded` and if it's not then (`promiseResult.Asset` is null) log an error

File: Explorer/Assets/DCL/SDKComponents/SceneUI/Systems/UIDropdown/UIDropdownReleaseSystem.cs [hunk 3]
```diff
@@ -24,7 +24,7 @@ public partial class UIDropdownReleaseSystem : BaseUnityLoopSystem
 
         private UIDropdownReleaseSystem(World world, IComponentPoolsRegistry poolsRegistry) : base(world)
         {
-            poolsRegistry.TryGetPool(typeof(DCLDropdown), out componentPool);
+            poolsRegistry.TryGetPool(typeof(UIDropdownComponent), out componentPool);
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/SDKComponents/SceneUI/Systems/UIDropdown/UIDropdownReleaseSystem.cs:27]: I think you can use `GetReferenceTypePool<T>` instead and don't expect it to return `null`. In general if you didn't register the pool of this type for some reason it should be detected as soon as possible and not linger silently

**Review by mikhail-dcl — CHANGES_REQUESTED** [review]
Looks good to me, I just have several proposes for fixes

File: Explorer/Assets/DCL/SDKComponents/SceneUI/Systems/UIBackground/UIBackgroundInstantiationSystem.cs (hunk 2 above)
Comment (sandrade-dcl) [reply Explorer/Assets/DCL/SDKComponents/SceneUI/Systems/UIBackground/UIBackgroundInstantiationSystem.cs]: Done: https://github.com/decentraland/unity-explorer/pull/303/commits/687b1a9f58f5ecd9c26a0b5d89854c69e072069d

File: Explorer/Assets/DCL/SDKComponents/SceneUI/Systems/UIDropdown/UIDropdownReleaseSystem.cs (hunk 3 above)
Comment (sandrade-dcl) [reply Explorer/Assets/DCL/SDKComponents/SceneUI/Systems/UIDropdown/UIDropdownReleaseSystem.cs]: Done: https://github.com/decentraland/unity-explorer/pull/303/commits/56cc6f24969c036b6363ad201a03c416618f94a1

**Review by mikhail-dcl — APPROVED** [review]
Looks great, thanks 💚

**Pull request merged**